File: app/src/main/java/com/phoenix/phoenix/entity/Merce.java
package com.phoenix.phoenix.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@Entity
@Table(name = "merce")
public class Merce {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String nome;

    @Enumerated(EnumType.STRING)
    private TipoMerce tipo;

    private double prezzo;

    private int quantita;

    @ManyToOne
    @JoinColumn(name = "fornitura")
    @JsonIgnore
    private Fornitura fornitura;

}

File: front/src/app/model/merce.ts

export interface Merce{
    id:number;
    nome:string;
    tipo:string;
    prezzo:number;
    quantita:number;
}

export class Merce{
    constructor(
        public nome:string,
        public tipo:string,
        public prezzo:number,
        public quantita:number
    ) {}
}